### PR TITLE
Fix on active model secure password update to an empty string("") not working properly

### PR DESCRIPTION
**Pull request by rajesh-rrk**
Reference - https://github.com/rails/rails/issues/44516

### CODE REVIEW

File: activemodel/lib/active_model/secure_password.rb [hunk 1]
```diff
@@ -93,10 +93,10 @@ def initialize(attribute)
         attr_reader attribute
 
         define_method("#{attribute}=") do |unencrypted_password|
-          if unencrypted_password.nil?
+          if unencrypted_password.nil? || unencrypted_password.empty?
```
Comment (mansakondo) [inline activemodel/lib/active_model/secure_password.rb:96]: Ah you didn't use `unencrypted_password.blank?` after all.
```suggestion
          if unencrypted_password.blank?
```

Comment (rajesh-rrk) [reply activemodel/lib/active_model/secure_password.rb]: oops, I was updated with `blank?` now :)

**Comment by eileencodes** [comment]
Hi @rajesh-rrk thanks for the PR. According to the tests the previous behavior is correct. See https://github.com/rails/rails/blob/main/activemodel/test/cases/secure_password_test.rb#L126-L130 and test failures in https://buildkite.com/rails/rails/builds/85018#c042d378-7fcc-4cec-896f-5f19af32439c/1179-1187.

The password when blank isn't updated and is silently ignored but doesn't throw an error.

**Comment by fabian12943** [comment]
Is there a reason why this PR never got merged into main? The current behavior is still broken.

**Pull request closed without merging**